jax/_src/test_util.py

```diff
@@ -1452,7 +1453,8 @@ def integer(self):
 
   @_cached_property
   def all_integer(self):
-    return self.supported([
+    new_dtypes = [_dtypes.int2] if _dtypes.int2 is not None else []
+    return self.supported(new_dtypes + [
```

Comment:
It looks like neither `int2` nor `int4` is actually making it to the tests, because neither is is listed under the `supported_dtypes` above. You can try adding them to the supported types, but I suspect this would lead to many many test failures, because these are not actually supported for many operations across the package.

I think the best bet is to revert changes here, and then directly test `int2` and `int4` sums in the test file.